python/pyarrow/array.pxi

```diff
@@ -1475,6 +1475,17 @@ cdef class Array(_PandasConvertible):
         return self.take(indices)
 
     def _to_pandas(self, options, types_mapper=None, **kwargs):
+        pandas_dtype = None
+        try:
+            pandas_dtype = self.type.to_pandas_dtype()
```

Comment:
Something that we should also do the the other places where this logic lives (eg ChunkedArray._to_pandas), but I think we should give precedence to `types_mapper`, if specified. Because as a user, if you specify that keyword, you want to be able to override the default conversion.